osu.Game.Rulesets.Catch/Difficulty/CatchPerformanceCalculator.cs

```diff
@@ -66,25 +65,40 @@ public override double Calculate(Dictionary<string, double> categoryDifficulty =
             if (approachRate > 10.0)
                 approachRateFactor += 0.1 * (approachRate - 10.0); // Additional 10% at AR 11, 30% total
             else if (approachRate < 8.0)
-                approachRateFactor += 0.025 * (8.0 - approachRate); // 2.5% for each AR below 8
+                approachRateFactor += 0.04 * (8.0 - approachRate); // 4% for each AR below 8
 
             value *= approachRateFactor;
 
             if (mods.Any(m => m is ModHidden))
             {
                 // Hiddens gives almost nothing on max approach rate, and more the lower it is
                 if (approachRate <= 10.0)
-                    value *= 1.05 + 0.075 * (10.0 - approachRate); // 7.5% for each AR below 10
+                    value *= 1.05 + 0.10 * (10.0 - Math.Min(10.0, approachRate)); // 10% for each AR below 10
                 else if (approachRate > 10.0)
                     value *= 1.01 + 0.04 * (11.0 - Math.Min(11.0, approachRate)); // 5% at AR 10, 1% at AR 11
+
+                if (approachRate <= 9.0)
+                    value *= 1 + 0.04 * (8.0 - approachRate); // Additional 4% for each AR below 9
             }
 
             if (mods.Any(m => m is ModFlashlight))
+            {
                 // Apply length bonus again if flashlight is on simply because it becomes a lot harder on longer maps.
                 value *= 1.35 * lengthBonus;
 
+                if (approachRate > 8.0f)
+                    value *= 0.1f * (approachRate - 8.0) + 1; // 10% for each AR above 8
+
+                if (approachRate < 8.0f)
+                    value *= 0.06f * (8.0 - approachRate) + 1; // -6% for each AR below 8
+            }
+
             // Scale the aim value with accuracy _slightly_
-            value *= Math.Pow(accuracy(), 5.5);
+            value *= Math.Pow(accuracy(), 6);
+
+            // Custom multiplier for HalfTime -> slower catcher = easier to control
+            if (mods.Any(m => m is ModHalfTime))
```

Comment:
It was supposed to apply to both HT and DT but I removed DT because DT's SR value is already pretty high with my changes. I think I'm going to remove HT to keep a clean logic.

Reply:
Note that you don't have to change everything mentioned in my comments (because it is just a question).
Though your reasoning (ad-hoc nerf) was weak in this case so it probably can be removed (and potentially revisit later) IMO.

Reply:
I just added a thing to my testing software to properly test HT and I noticed that the Star Rating with HalfTime is better than what I expected. And it makes much more sense to have no penalty given for HalfTime if DoubleTime is not buffed. As you said, the logic is flawed.